api-gateway/src/main/kotlin/com/saveourtool/save/gateway/utils/AuthorizationHeadersGatewayFilterFactory.kt

```diff
@@ -0,0 +1,55 @@
+package com.saveourtool.save.gateway.utils
+
+import com.saveourtool.save.authservice.utils.AuthenticationUserDetails
+import com.saveourtool.save.gateway.service.BackendService
+import com.saveourtool.save.utils.switchIfEmptyToResponseException
+import org.springframework.cloud.gateway.filter.GatewayFilter
+import org.springframework.cloud.gateway.filter.GatewayFilterChain
+import org.springframework.cloud.gateway.filter.factory.AbstractGatewayFilterFactory
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpStatus
+import org.springframework.security.authentication.BadCredentialsException
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken
+import org.springframework.stereotype.Component
+import org.springframework.web.server.ServerWebExchange
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.toMono
+import java.security.Principal
+
+/**
+ * Filter, that mutate existing exchange,
+ * inserts user's info into Authorization headers instead of existing value, not paying attention to the credentials,
+ * since at this moment they are already checked by gateway.
+ */
+@Component
+class AuthorizationHeadersGatewayFilterFactory(
+    private val backendService: BackendService,
+) : AbstractGatewayFilterFactory<Any>() {
+    override fun apply(config: Any?): GatewayFilter = GatewayFilter { exchange: ServerWebExchange, chain: GatewayFilterChain ->
+        exchange.getPrincipal<Principal>()
+            .flatMap { resolveSaveUser(it) }
+            .map { user ->
+                exchange.mutate()
+                    .request { builder ->
+                        builder.headers { headers: HttpHeaders ->
+                            headers.remove(HttpHeaders.AUTHORIZATION)
```

Comment:
btw - why do you remove authorization headers? 
May be not create it initially for particular endpoints?

Reply:
we don't use authorization header in backend. I just cleanup it to be sure that we don't pass through this value from external request